pycloudlib/gce/cloud.py

```diff
@@ -59,6 +61,17 @@ def __init__(
         self.region = region
         self.zone = '%s-%s' % (region, zone)
         self.instance_counter = count()
+        self._set_service_account_email()
+
+    def _set_service_account_email(self):
+        """Set service account email if credentials provided."""
+        json_credentials = {}
+
+        if self.credentials_path:
+            with open(self.credentials_path, 'r') as f:
+                json_credentials = json.load(f)
+
+        self._service_account_email = json_credentials.get("client_email")
```

Comment:
My `application_default_credentials.json` does not have a `client_email` key.

More broadly, I'm concerned about us relying on the credentials JSON: we've seen before that it doesn't have a predictable interface, so code written to use it is going to be unreliable.